README.md
### Welcome to the FIFA 23 Live Editor!

- [Changes in latest version of the tool](https://raw.githubusercontent.com/xAranaktu/FIFA-23-Live-Editor/main/changelog.txt)
- [FIFA 23 Live Editor & Cheat Table Development Board](https://bit.ly/F23-LE-CT-DEV-BOARD)
- [FIFA 23 Live Editor & Cheat Table Release Schedule](https://bit.ly/F23-Projects-Release-Schedule)
- Check [WIKI](https://github.com/xAranaktu/FIFA-23-Live-Editor/wiki) for Guides and Tutorials

### Read Me Before You Pledge or Buy The Game

This year the game is protected with brand new EA Anticheat. Using the tool may flag your account and EA may ban it later on.
You are using the tool at your own risk.

- The tool is at early stage of developement, many features may not work or may be bugged. Wait until at least until December if you expect somewhat finished tool
- If you are scared of being banned then don't use the tool (at least for a while until, situation will clarify)
- This tool doesn't work online. You won't be able to connect with EA servers while using it. This is intended, and won't be "fixed".
- I can't guarantee for how long the tool will be working. EA may update the game at any time and since then it may be not possible to update the Live Editor to make it compatible with latest game update.

***

If you want to download the tool then DONT CLONE THE REPO...
Get it from releases:

![](https://i.imgur.com/ocPeJiV.png)

***

![](https://i.imgur.com/W6lPg65.jpeg)
![](https://i.imgur.com/o1naouS.png)
![](https://i.imgur.com/0grWO3A.png)
![](https://i.imgur.com/4JGPfWK.png)
![](https://i.imgur.com/pwfIyfs.png)

![GitHub all releases](https://img.shields.io/github/downloads/xAranaktu/FIFA-23-Live-Editor/total) ![GitHub release (latest by date)](https://img.shields.io/github/downloads/xAranaktu/FIFA-23-Live-Editor/latest/total)


changelog.txt
v23.1.3.6:
    - Works with Title Update 17.1

v23.1.3.5:
    - Works with Title Update 17
    - Some of the message boxes now have "Don't show this message box anymore" checkbox.
    - Added "HKEY_LOCAL_MACHINE\SOFTWARE\Live Editor\EAFC 24\Dir" key to win registry which points to the folder where live editor dll is located.

v23.1.3.4:
    - Added:
        - AcceleRATE column to player list
        - AcceleRATE to player profile (No, you can't edit it)
        - AcceleRATE player advanced filter
        - Overall Rating player advanced filter
        - "Generate Youth Academy Report" to "Misc Features" -> "Career Mode" -> "Youth Academy"
    - Fixed "Goals For" and "Goals Against" in Standings Away tab.
    - Removed "Disable Chants" option from the settings

v23.1.3.3:
    - Works with Title Update 15

v23.1.3.2:
    - Fixed problem with distorted chants
    - Added 100+ accessories to the accessory list in player editor

v23.1.3.1:
    - Works with Title Update 14.1 

v23.1.3.0:
    - Restored previous version of the launcher, so the fake EAAC should be copied properly when you run the game via MM

v23.1.2.9:
    - Works with Title Update 14
    - Fixed bug causing unability to change directiories in settings

v23.1.2.8:
    - Added to "Misc Features" -> "Gameplay" -> "Perfect Timed Shot"
    - Changed the way that Live Editor adds players to teams, this should solve the issue with player not appearing in team after promotion from YA, or transfer.
    - Fixed game crash caused by created job offer.
    - Fixed transfer bans not being saved properly.

v23.1.2.7:
    - Updated LUA from `5.3.4 (32 bit)` to `5.4.6 (x64)`
    - Locale.ini window is now hidden by default to reduce the false reports about bug with "key file not found"
    - In settings you can now define where your game is installed (to fix launcher error code 2)
    - Added: 
        - "Reveal Player data" in "Misc Features"
        - "LiveEditorClearRegistry.reg" file to the live editor folder. (it clears windows registry from Live Editor related keys)
        - Option to generate job offer in manager career mode from any club/national team (you can find it in team editor).
    - Fixed:
        - Lacking name of the created club
        - Players not appearing in their new club after manual transfer

v23.1.2.6:
    - Works with Title Update 12
    - Added to Players Editor:
        - Other -> Team selection bias
        - Other -> Team selection bias position
    - Added to Formation Editor:
        - Offense -> Build Up Play
        - Offense -> Chance Creation
        - Freeze Lineup
    - Removed from Formation Editor:
        - Offense -> Offensive Style (seems to be no longer in use by FIFA 23)
    - Live Editor will now update cached formations in career mode when you change formation in Formation Editor.

v23.1.2.5:
    - Live Editor settings can now also be changed from the launcher.
    - Added "locale.ini" editor in the launcher.
    - Added option allowing for player name change in player editor.

v23.1.2.4:
    - You can now perform search by name or id by clicking "Enter" key, instead of mouse clicking the magnifier icon.
    - Fixed problems with Live Editor Mods Directory (hopefully for good)
    - Changed default directory for mods and other LE data from "<YOUR_GAME_INSTALLATION_DIR>\LiveEditorMods" to "C:\FIFA 23 Live Editor"
    - config.json is now stored in "C:\FIFA 23 Live Editor" instead of the LE folder to keep your settings even if you update the tool.

v23.1.2.3:
    - Works with Title Update 10

v23.1.2.2:
    - Fixed game crash that were occurring on slower machines while waiting for game window.
    - Changed default injection delay to 100ms

v23.1.2.1:
    - Added "Misc Features" with:
        - Unsackable
        - Club Transfer Budget
    - Added support for file paths containing UTF-16 characters (not fully tested).
    - Searching players/teams by name will now also match accented characters. (For example "makelele" will match "Makélélé")
    - Fixed artificialkey field error in "CreatePlayer" lua function
    - Reduced GPU & Power usage by the launcher.

v23.1.2.0:
    - It's possible now to define custom score for favourite and disliked teams.
    - Added advanced filters for players and teams.

v23.1.1.9:
    - Added to Teams Editor:
        - Stadium
        - Kits

v23.1.1.8:
    - Added Fixtures to Teams Editor (Match-fixing & Rescheduling Games)
    - Fixed caching offsets, minor game updates shouldn't break the tool forcing you to redownload anymore.

v23.1.1.7:
    - Added to Teams Editor:
        - Transfer Bans
        - Formation Editor

v23.1.1.6:
    - Works with Title Update 6
    - Added 2 new traits to players editor, First Time Shots & Round the Keeper (thanks tokke001)
    - Added Player OVR column in Transfer History list

v23.1.1.5:
    - Added "Transfer History" with option to undo unwanted transfers.

v23.1.1.4:
    - Added "Teams" and "Team Editor" with:
        - Basic info
        - Players
        - Manager
        - Standings

v23.1.1.3:
    - Added "Season Statistics" to player editor

v23.1.1.2:
    - Fixed blacksceen issue.
    - Launcher:
        - Fixed "Error access denied" (maybe, lol)

v23.1.1.1:
    - Fixed:
        - Filtering values in the edit option popup
    - Launcher:
        - Fixed auto inject delay
        - Fixed "Error access denied" when starting the game manually

v23.1.1.0:
    - Added:
        - Lua Engine
    - Fixed:
        - Editing player jersey number

v23.1.0.9:
    - Launcher:
        - Fixed editing injection delay when using auto inject
        - Increased default injection delay from 300ms to 3000ms

v23.1.0.8:
    - Works with Title Update 4
    - UI Changes:
        - Replaced combo input with searchable input
    - Fixed:
        - Crash when loading player that doesn't exists, but have teamplayerlink

v23.1.0.7:
    - Added:
        - Option to transfer players
        - Option to loan players

lua/scripts/export_season_stats.lua
-- This script will export current season stats (goals scored, assists etc.) to CSV.
-- The SEASON_STATS.csv file will be created on Desktop.
-- The game resets stats at the beginning of every season, so you need to run the script at the end of every season.

local desktop_path = string.format("%s\\Desktop", os.getenv('USERPROFILE'))
local current_date = GetCurrentDate()
local file_path = string.format("%s\\SEASON_STATS_%d_%d_%d.csv", desktop_path, current_date.day, current_date.month, current_date.year)

-- Don't Touch
local columns = {
    "position",
    "playerid",
    "playername",
    --"team",
    "competition",
    "appearances",
    "AVG",
    "MOTMs",
    "goals",
    "assists",
    "yellow_cards",
    "two_yellow",
    "red_cards",
    "saves",
    "goals_conceded",
    "cleansheets",
}

local col_to_key = {
    position = "position",
    playerid = "playerid",
    playername = "playername",
    --team = "teamname",
    competition = "compname",
    appearances = "app",
    AVG = "avg",
    MOTMs = "motm",
    goals = "goals",
    assists = "assists",
    yellow_cards = "yellow",
    two_yellow = "two_yellow",
    red_cards = "red",
    saves = "saves",
    goals_conceded = "goals_conceded",
    cleansheets = "clean_sheets",
}

function get_pos_name(pos_id)
    POS_TO_NAME = {
        [0] = "GK",
        [1] = "SW",
        [2] = "RWB",
        [3] = "RB",
        [4] = "RCB",
        [5] = "CB",
        [6] = "LCB",
        [7] = "LB",
        [8] = "LWB",
        [9] = "RDM",
        [10] = "CDM",
        [11] = "LDM",
        [12] = "RM",
        [13] = "RCM",
        [14] = "CM",
        [15] = "LCM",
        [16] = "LM",
        [17] = "RAM",
        [18] = "CAM",
        [19] = "LAM",
        [20] = "RF",
        [21] = "CF",
        [22] = "LF",
        [23] = "RW",
        [24] = "RS",
        [25] = "ST",
        [26] = "LS",
        [27] = "LW"
    }
    return POS_TO_NAME[pos_id] or "INVALID"
end

function get_players()
    local result = {}
    
    local rows = GetDBTableRows("players")
    for i=1, #rows do
        local row = rows[i]
        local playerid = math.floor(row.playerid.value)
        
        local player = {}
        
        player.position = get_pos_name(math.floor(row.preferredposition1.value))
        
        -- Get the names later, only if needed as it's very slow
        player.name = ""
        --player.teamname = ""

        result[playerid] = player
    end
    
    return result
end

function get_stats(players)
    local result = {}
   
    local all_stats = GetPlayersStats()

    for i=1, #all_stats do
        local stat = all_stats[i]
        local playerid = stat.playerid
        local app = stat.app

        if playerid > 0 and playerid < 4294967295 and app > 0 then
            local player = players[playerid]
            if player ~= nil then
                if player.name == "" then
                    player.name = GetPlayerName(playerid)
                end
                
                local stats = player.stats or {}
                local compobjid = stat.compobjid
                stats[compobjid] = {}
                
                stats[compobjid]["playerid"] = playerid
                stats[compobjid]["playername"] = player.name
                stats[compobjid]["position"] = player.position
                
                stats[compobjid]["goals"] = stat.goals
                stats[compobjid]["yellow"] = stat.yellow
                stats[compobjid]["red"] = stat.red
                stats[compobjid]["assists"] = stat.assists
                stats[compobjid]["clean_sheets"] = stat.clean_sheets
                stats[compobjid]["compobjid"] = compobjid
                stats[compobjid]["compname"] = GetCompetitionNameByObjID(compobjid)
                
                -- Not sure
                stats[compobjid]["motm"] = stat.motm
                stats[compobjid]["saves"] = stat.saves
                stats[compobjid]["goals_conceded"] = stat.goals_conceded
                stats[compobjid]["two_yellow"] = stat.two_yellow
                
                local avg = stat.avg
                if app > 1 then 
                    avg = (avg / app) / 10 
                elseif app == 1 then
                    avg = avg / 10
                end
                
                stats[compobjid]["app"] = app
                stats[compobjid]["avg"] = string.format("%0.2f", avg)

                player.stats = stats
                result[playerid] = player

            end
        end
    end

    return result
end

assert(IsInCM(), "Script must be executed in career mode")

local file = io.open(file_path, "w+")
io.output(file)

io.write(table.concat(columns, ","))
io.write("\n")
local players_stats = get_stats(get_players())
for playerid, data in pairs(players_stats) do
    for compobjid, stat in pairs(data.stats) do
        local row = {}
        for i=1, #columns do
            local colname = columns[i]
            local _key = col_to_key[colname]
            table.insert(row, stat[_key])
        end
        io.write(table.concat(row, ","))
        io.write("\n")
    end

end

io.close(file)
Log("Done")
